WordPress/Classes/Models/SharingButton+Lookup.swift

```diff
@@ -0,0 +1,32 @@
+extension SharingButton {
+
+    /// Returns an array of all cached `SharingButtons` objects.
+    ///
+    /// - Parameters
+    ///     - blog: A `Blog` object
+    ///
+    /// - Returns: An array of `SharingButton`s.  The array is empty if no objects are cached.
+    ///
+    @objc(allSharingButtonsForBlog:inContext:error:)
+    static func allSharingButtons(for blog: Blog, in context: NSManagedObjectContext) throws -> [SharingButton] {
+        let request = NSFetchRequest<SharingButton>(entityName: SharingButton.classNameWithoutNamespaces())
+        request.predicate = NSPredicate(format: "blog = %@", blog)
+        request.sortDescriptors = [NSSortDescriptor(key: "order", ascending: true)]
+        return try context.fetch(request)
+    }
+
+    /// Finds a cached `SharingButton` by its `buttonID` for the specified `Blog`
+    ///
+    /// - Parameters:
+    ///     - buttonID: The button ID of the `sharingButton`.
+    ///     - blog: The blog that owns the sharing button.
+    ///
+    /// - Returns: The requested `SharingButton` or nil.
+    ///
+    static func lookupSharingButton(byID buttonID: String, for blog: Blog, in context: NSManagedObjectContext) throws -> SharingButton? {
+        let request = NSFetchRequest<SharingButton>(entityName: SharingButton.classNameWithoutNamespaces())
+        request.predicate = NSPredicate(format: "buttonID = %@ AND blog = %@", buttonID, blog)
+        return try context.fetch(request).first
+    }
```

Comment:
Wow... I was not expecting a button to be stored in the database 🤔 

I guess there are a few properties worth storing, but still...

https://github.com/wordpress-mobile/WordPress-iOS/blob/f5c8424947372018f487109666c9319bb8261cec/WordPress/Classes/Models/SharingButton.swift#L12-L18

Reply:
Yeah, it's pretty common in the app actually.

There are roughly three kinds of database records (a.k.a the managed objects) in the app.
1. The database records are actually useful and must not be discarded. Like `Post` for local drafts.
2. The database records are just a cache, which is used to display UI initially but refreshed later. Like, media objects in the WordPress media library.
3. The database records are saved for the convenience of displaying them using a `NSFetchedResultController`. These records can be deleted immediately after the user exists the corresponding screen, without affecting the app. Like, sharing buttons in this PR, and "managed people" in "People" screens, some posts in the Reader tab, etc.

My feeling is, the third kind is the majority.

Reply:
> The database records are saved for the convenience of displaying them using a `NSFetchedResultController`. These records can be deleted immediately after the user exists the corresponding screen, without affecting the app.

Is the convenience of displaying them in this way worth the overhead?

Reply:
That's not an easy question to answer, I think 🥲 .

Here is another way to look at it. It's a super common pattern in the app to use a `NSFetchedResultController` to display a list of things. It's so common that we have a [`WPTableViewHandler`](https://github.com/wordpress-mobile/WordPress-iOS/blob/82d6af6b137598ef235a6860c04a00c408be3846/WordPress/Classes/Utility/WPTableViewHandler.h) for this exact purpose.

Now, here comes my assumption. When we start building a new feature that displays a list of stuff, I think the following is probably a part of the mindset. Naturally we would go with the existing patterns, the most common one, which is using a `NSFetchedResultController`. Then we'll see what it requires. It requires a managed object? Okay, I'll add a new model to the Core Data model file then. That's probably how we end up with so many the third kind of model types I mentioned above.

And a question in my head is, when it comes to implementing screens like the "People" or the "Menus" screen (which both uses the third kind of model types I mentioned above—`ManagedPerson` and `Menu` are not used anywhere else in the app and the records stored in the database aren't used), do we need to create a Core Data model for it? Obviously, we surely can build them without using Core Data. We just don't have alternative to consider (maybe there is that I am not aware of).

Reply:
Thank you, this is super insightful. Sounds like a matter of path dependance and inertia. @alpavanoglu and I talked briefly privately about something similar in regards to adopting SwiftUI. There are already well established patterns that serve us well enough, sticking with them is often the path of least resistance.